### PR TITLE
fix(block-reverter): Fix reverting snapshot files

### PR DESCRIPTION
**Pull request by slowli**
## What ❔

- Does not retry "not found" errors when removing objects.
- Makes rolling back snapshot files optional.

## Why ❔

The current implementation leads to very slow reverts because of retries.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `zk fmt` and `zk lint`.
- [x] Spellcheck has been run via `zk spellcheck`.

### CODE REVIEW

File: core/lib/object_store/src/gcs.rs [hunk 1]
```diff
@@ -17,31 +17,39 @@ use google_cloud_storage::{
...
 {
     let mut retries = 1;
-    let mut backoff = 1;
+    let mut backoff_secs = 1;
     loop {
         match f().await {
             Ok(result) => return Ok(result),
-            Err(err) => {
-                tracing::warn!(%err, "Failed GCS request {retries}/{max_retries}, retrying.");
+            Err(err) if err.is_transient() => {
```
Comment (slowli) [inline core/lib/object_store/src/gcs.rs:37]: This change can lead to issues where `ObjectStore` previously relied on retries for logically non-transient errors (e.g., "not found" errors). Do we want to change it, or use a safer approach (e.g., consider all errors transient / retryable unless removal is concerned)?

File: core/node/block_reverter/src/lib.rs [hunk 2]
```diff
@@ -380,6 +380,8 @@ impl BlockReverter {
         object_store: &dyn ObjectStore,
         deleted_snapshots: &[SnapshotMetadata],
     ) -> anyhow::Result<()> {
+        const CONCURRENT_REMOVE_REQUESTS: usize = 20;
```
Comment (slowli) [inline core/node/block_reverter/src/lib.rs:383]: Is this a reasonable value? Unfortunately, I haven't managed to find "native" batching for GCS API (it looks like classic REST API, so it's not quite unexpected).

File: core/node/block_reverter/src/lib.rs [hunk 3]
```diff
@@ -419,18 +421,46 @@ impl BlockReverter {
...
+                        if !is_incomplete_snapshot {
+                            is_incomplete_snapshot = true;
+                            tracing::warn!(
+                                "Snapshot for L1 batch #{} is incomplete (misses al least storage logs chunk ID {chunk_id}). \
+                                 It is probable that it's currently being created, in which case you'll need to clean up produced files \
+                                 manually afterwards",
+                                snapshot.l1_batch_number
+                            );
+                        }
+                        return None;
```
Comment (slowli) [inline core/node/block_reverter/src/lib.rs:438]: Maybe, it does make sense to attempt removing "future" storage log chunks?

File: core/lib/object_store/src/gcs.rs (hunk 1 above)
Comment (popzxc) [reply core/lib/object_store/src/gcs.rs:37]: I think the change makes total sense. If it occurs that it breaks something (I _really_ hope it does not), we can simply revert the commit and reimplement the change with the newly obtained knowledge.

File: core/node/block_reverter/src/lib.rs (hunk 2 above)
Comment (RomanBrodetski) [reply core/node/block_reverter/src/lib.rs:383]: I think it's reasonable

File: core/node/block_reverter/src/lib.rs (hunk 3 above)
Comment (RomanBrodetski) [reply core/node/block_reverter/src/lib.rs:438]: I think it doesn't - snapshot creator shouldn't be running while a rollback is taking place. Let's keep as is and remember/revisit when we revamping the block revertion next wave

**Review by RomanBrodetski — APPROVED** [review]

**Pull request merged**